platooning_strategic_IHP/src/platoon_strategic_ihp_node.cpp

```diff
@@ -0,0 +1,241 @@
+
+
+    // Setup subscribers
+    mob_request_sub = create_subscription<carma_v2x_msgs::msg::MobilityRequest>("incoming_mobility_request", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::mob_req_cb,  worker_.get(), std_ph::_1));
+
+    mob_response_sub = create_subscription<carma_v2x_msgs::msg::MobilityResponse>("incoming_mobility_response", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::mob_resp_cb, worker_.get(), std_ph::_1));
+
+    mob_operation_sub = create_subscription<carma_v2x_msgs::msg::MobilityOperation>("incoming_mobility_operation", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::mob_op_cb, worker_.get(), std_ph::_1));
+
+    current_pose_sub = create_subscription<geometry_msgs::msg::PoseStamped>("current_pose", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::pose_cb, worker_.get(), std_ph::_1));
+
+    current_twist_sub = create_subscription<geometry_msgs::msg::TwistStamped>("current_velocity", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::twist_cb, worker_.get(), std_ph::_1));
+
+    cmd_sub = create_subscription<geometry_msgs::msg::TwistStamped>("twist_raw", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::cmd_cb, worker_.get(), std_ph::_1));
+
+    georeference_sub = create_subscription<std_msgs::msg::String>("georeference", 10,
+                                                              std::bind(&PlatoonStrategicIHPPlugin::georeference_cb, worker_.get(), std_ph::_1));
+
+    loop_timer_ = create_timer(
+        get_clock(),
+        std::chrono::milliseconds(100), // 10 Hz frequency
+        std::bind(&PlatoonStrategicIHPPlugin::onSpin, worker_.get()));
+
+    // Return success if everything initialized successfully
+    return CallbackReturn::SUCCESS;
+  }
+
+  carma_ros2_utils::CallbackReturn Node::on_cleanup_plugin()
+  {
+    // Ensure subscribers are disconnected incase cleanup is called, we don't want to keep driving the worker
+    mob_response_sub.reset();
+    mob_operation_sub.reset();
+    current_pose_sub.reset();
+    current_twist_sub.reset();
+    cmd_sub.reset();
+    georeference_sub.reset();
+    worker_.reset();
+
+    return CallbackReturn::SUCCESS;
+  }
+
+
+  void Node::plan_maneuvers_callback(
+    std::shared_ptr<rmw_request_id_t>, 
+    carma_planning_msgs::srv::PlanManeuvers::Request::SharedPtr req, 
+    carma_planning_msgs::srv::PlanManeuvers::Response::SharedPtr resp)
+  {
+    if (worker_)
+      worker_->plan_maneuver_cb(*req, *resp);
+  }
+
+  bool Node::get_availability() {
+    return true;
+  }
+
+  std::string Node::get_version_id() {
+    return "v4.0";
```

Comment:
Should this be a configurable parameter?

Reply:
I don't think so, version numbers should always be impossible for users to change. If it was a configuration parameter, I could make no code changes, but pretend to run an older or newer version.